From: YOKOTA Hiroshi <yokota.hgml@gmail.com>
Date: Mon, 1 Jul 2024 15:49:53 +0900
Subject: Fix regexp escape sequence syntax warning (Closes: #1074559)

Use "r" prefix to use raw strings.
It reduces unwanted backslash escapement in literal string parser.

See also "re" module document in the Python Standard Library.

----
while installing calibre this warning comes up:

Configuring calibre (7.13.0+ds-1)...
/usr/lib/calibre/templite/__init__.py:35: SyntaxWarning: invalid escape
sequence '\['
  auto_emit = re.compile('(^[\'\"])|(^[a-zA-Z0-9_\[\]\'\"]+$)')
----
---
 src/templite/__init__.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/templite/__init__.py b/src/templite/__init__.py
index da5c72a..5c39b49 100644
--- a/src/templite/__init__.py
+++ b/src/templite/__init__.py
@@ -32,7 +32,7 @@ from polyglot.builtins import unicode_type
 
 
 class Templite:
-    auto_emit = re.compile('(^[\'\"])|(^[a-zA-Z0-9_\[\]\'\"]+$)')
+    auto_emit = re.compile(r'(^[\'\"])|(^[a-zA-Z0-9_\[\]\'\"]+$)')
 
     def __init__(self, template, start='${', end='}$'):
         if len(start) != 2 or len(end) != 2:
